File: fixtures/python/sample.py
# Simple function
def simple():
    return 42

# Function with decision points
def complex_func(x, y):
    if x > 0:
        for i in range(x):
            if i % 2 == 0:
                y += i
            elif i % 3 == 0:
                y -= i
            else:
                y *= 2
    elif x < 0:
        while y > 0:
            y -= 1
    else:
        try:
            y = x / y
        except ZeroDivisionError:
            y = 0
    return y

# Function with deep nesting
def deeply_nested(data):
    result = []
    for item in data:
        if item is not None:
            if isinstance(item, list):
                for sub in item:
                    if sub > 0:
                        result.append(sub)
            elif isinstance(item, dict):
                for key, value in item.items():
                    if value:
                        result.append(key)
    return result

# Class with methods
class Calculator:
    def add(self, a, b):
        return a + b

    def compute(self, operation, a, b):
        if operation == "add":
            return self.add(a, b)
        elif operation == "sub":
            return a - b
        elif operation == "mul":
            return a * b
        elif operation == "div":
            if b != 0:
                return a / b
            else:
                raise ValueError("Division by zero")
        else:
            raise ValueError(f"Unknown operation: {operation}")

File: fixtures/rust/sample.rs
// Simple function
fn simple() -> i32 {
    42
}

// Complex function
fn complex_func(x: i32, mut y: i32) -> i32 {
    if x > 0 {
        for i in 0..x {
            if i % 2 == 0 {
                y += i;
            } else if i % 3 == 0 {
                y -= i;
            } else {
                y *= 2;
            }
        }
    } else if x < 0 {
        while y > 0 {
            y -= 1;
        }
    } else {
        if y != 0 {
            y = x / y;
        } else {
            y = 0;
        }
    }
    y
}

// Function with match
fn classify(x: i32) -> &'static str {
    match x {
        x if x > 100 => "large",
        x if x > 10 => "medium",
        x if x > 0 => "small",
        _ => "negative",
    }
}

// Struct with methods
struct Calculator;

impl Calculator {
    fn compute(&self, op: &str, a: i32, b: i32) -> Result<i32, String> {
        match op {
            "add" => Ok(a + b),
            "sub" => Ok(a - b),
            "mul" => Ok(a * b),
            "div" => {
                if b != 0 {
                    Ok(a / b)
                } else {
                    Err("Division by zero".to_string())
                }
            }
            _ => Err(format!("Unknown operation: {}", op)),
        }
    }
}

fn main() {
    println!("{}", simple());
}
